### ISSUE TITLE
Jenkins Plugin

### ISSUE THREAD

**Issue by vanniktech** [shyiko/ktlint#66]
It'd be super cool if there would be a Jenkins Plugin for ktlint. I also believe that the core functionality already exists since Checkstyle, PMD & Findbugs Plugin all show the errors the same way with the same information.

Maybe it's as simple as parsing the report and transforming it to the Jenkins structure.

**Comment by shyiko** [comment]
@vanniktech I believe this will (is) taken care of in the upcoming release (which has a built-in checkstyle reporter (among others)). You'll be able to point Jenkins to the report file and everything should just work (provided I understand your request correctly).

**Comment by vanniktech** [comment]
It'll still need a Jenkins plugin though, right?

**Comment by shyiko** [comment]
Last time I used Jenkins Checkstyle plugin allowed to specify multiple checkstyle files, so as long as ktlint can produce a file in a checkstyle-compatible format - there is no need for a separate plugin.

**Comment by vanniktech** [comment]
Oh ok now I understand it. That would make sense yeah.

**Comment by shyiko** [comment]
`checkstyle` reporter added in [0.9.0](https://github.com/shyiko/ktlint/releases/tag/0.9.0) (`ktlint --reporter=checkstyle > ktlint-report-in-checkstyle-format.xml` should do the trick) (gradle & maven examples are available in the readme).

**Comment by vanniktech** [comment]
The generation does not quite work. `ktlint` fails with 

```
Missing spacing around "="
```

However when I execute `ktlintGenerateCheckstyleXML` it generates the following file:

```
<?xml version="1.0" encoding="utf-8"?>
<checkstyle version="8.0">
</checkstyle>
```

which has no violations.

**Comment by shyiko** [comment]
@vanniktech 
```sh
$ cd /tmp
$ echo "val x=1" > A.kt
$ ktlint A.kt 
/tmp/A.kt:1:6: Missing spacing around "="
$ ktlint A.kt --reporter=checkstyle
<?xml version="1.0" encoding="utf-8"?>
<checkstyle version="8.0">
	<file name="/tmp/A.kt">
		<error line="1" column="6" severity="error" message="Missing spacing around &quot;=&quot;" source="op-spacing" />
	</file>
</checkstyle>
```
Check that pattern is actually matching files (you can try adding --debug CLI argument).

**Comment by vanniktech** [comment]
The normal `ktlint` task flags it so I'd expect it to be a generated checkstyle violation.

**Comment by vanniktech** [comment]
Hmmm weird. Now it does seem to be working. If I can reproduce this safely again, I'll report back.
